identity/client/src/pages/roles/modals/useAllPermissionsTranslated.tsx

```diff
@@ -5,25 +5,20 @@
  * Licensed under the Camunda License 1.0. You may not use this file
  * except in compliance with the Camunda License 1.0.
  */
-import { ReactElement, useState } from "react";
+
 import { Role } from "src/utility/api/roles";
 import { default as allPermissions } from "src/configuration/permissions";
-import useTranslate from "src/utility/localization";
-
-type RoleListItem = {
-  permission: string;
-  description: ReactElement | string;
-};
+import { useState } from "react";
+import usePermissionsTranslated, {
+  TranslatedPermissionItem,
+} from "src/pages/roles/modals/usePermissionsTranslated";
 
-const usePermissions = (initialPermissions: Role["permissions"] = []) => {
-  const { t } = useTranslate("permissions");
+const useAllPermissionsTranslated = (
+  initialPermissions: Role["permissions"] = [],
+) => {
   const [permissions, setPermissions] = useState<string[]>(initialPermissions);
-  const availableItems: RoleListItem[] = allPermissions.map(
-    (permission: string) => ({
-      permission,
-      description: t(`${permission}.description`),
-    }),
-  );
+  const availableItems: TranslatedPermissionItem[] =
+    usePermissionsTranslated(allPermissions);
```

Comment:
There is no need to specify the type TranslatedPermissionItem because it is inferred from `usePermissionsTranslated`. Generally I would never specify a type if it is inferred and if it technically not needed to specify it. Typescript will still check all types and eslint ensures that we have no `any`s